mvtprovider/postgis/README.md

```diff
@@ -0,0 +1,131 @@
+- `max_connections` (int): [Optional] The max connections to maintain in the connection pool. Defaults to 100. 0 means no max.
+- `srid` (int): [Optional] The default SRID for the provider. Defaults to WebMercator (3857) but also supports WGS84 (4326)
+
+## Provider Layers
+
+In addition to the connection configuration above, Provider Layers need to be configured. A Provider Layer tells tegola how to query PostGIS for a certain layer. When using the PostGIS MVT Provider the `ST_AsMVTGeom()` MUST be used. An example minimum config using the `sql` config option:
+
+```toml
+[[mvt_providers.layers]]
+name = "landuse"
+# this table uses "geom" for the geometry_fieldname and "gid" for the id_fieldname so they don't need to be configured
+sql = "SELECT ST_AsMVTGeom(geom,!BBOX!) AS geom, gid FROM gis.landuse WHERE geom && !BBOX!"
+```
+
+### Provider Layers Properties
+
+- `name` (string): [Required] the name of the layer. This is used to reference this layer from map layers.
+- `geometry_fieldname` (string): [Optional] the name of the filed which contains the geometry for the feature. defaults to `geom`.
+- `id_fieldname` (string): [Optional] the name of the feature id field. defaults to `gid`.
+- `geometry_type` (string): [Optional] the layer geometry type. If not set, the table will be inspected at startup to try and infer the gemetry type. Valid values are: `Point`, `LineString`, `Polygon`, `MultiPoint`, `MultiLineString`, `MultiPolygon`, `GeometryCollection`.
+- `srid` (int): [Optional] the SRID of the layer. Supports `3857` (WebMercator) or `4326` (WGS84).
+- `sql` (string): [Required] custom SQL to use use. Supports the following tokens:
+  - `!BBOX!` - [Required] will be replaced with the bounding box of the tile before the query is sent to the database. `!bbox!` and`!BOX!` are supported as well for compatibilitiy with queries from Mapnik and MapServer styles.
+  - `!X!` - [Optional] will replaced with the "X" value of the requested tile.
+  - `!Y!` - [Optional] will replaced with the "Y" value of the requested tile.
+  - `!Z!` - [Optional] will replaced with the "Z" value of the requested tile.
+  - `!ZOOM!` - [Optional] will be replaced with the "Z" (zoom) value of the requested tile.
+  - `!SCALE_DENOMINATOR!` - [Optional] scale denominator, assuming 90.7 DPI (i.e. 0.28mm pixel size)
+  - `!PIXEL_WIDTH!` - [Optional] the pixel width in meters, assuming 256x256 tiles
+  - `!PIXEL_HEIGHT!` - [Optional] the pixel height in meters, assuming 256x256 tiles
+  - `!ID_FIELD!` - [Optional] the id field name
+  - `!GEOM_FIELD!` - [Optional] the geom field name
+  - `!GEOM_TYPE!` - [Optional] the geom type if defined otherwise ""
+
+## Example mvt_provider and map config
+
+**Important**: When referencing the `provider` in the `map` section of the config, you MUST prepend `mvt_` to the `provider_layer` value. This indicates to tegola that the provider is an MVT provider so tegola knows which provider section to perform the lookup. 
```

Comment:
Sorry if this has been talked about in person, or if I'm missing something, but it doesn't seem like the prefix is needed. When initializing `Map` both the `provider` and `mvtprovider` maps are checked:

https://github.com/go-spatial/tegola/blob/303488246a833a65a77e5fa0d0ee8029a187f0b3/cmd/internal/register/maps.go#L115-L118

Though, there needs to be an extra loop to check naming collisions like:
```go
// check name collisions between regular providers and mvtproviders
for k := range providers {
    if _, ok := mvtProviders[k]; ok {
        return provider.ErrProviderAlreadyExists { 
            Name: k,
        }
    }
}
```
before the following lines
https://github.com/go-spatial/tegola/blob/303488246a833a65a77e5fa0d0ee8029a187f0b3/cmd/tegola/cmd/root.go#L94
https://github.com/go-spatial/tegola/blob/303488246a833a65a77e5fa0d0ee8029a187f0b3/cmd/tegola_lambda/main.go#L86

Reply:
It's because of the config; if I remember correctly. And registering of the provider. We will eventually move to use the `type` field instead of having different sections. #688 tracks this.